pandas/tslib.pyx

```diff
@@ -3530,7 +3530,7 @@ def tz_convert(ndarray[int64_t] vals, object tz1, object tz2):
     if _get_zone(tz2) == 'UTC':
         return utc_dates
 
-    result = np.empty(n, dtype=np.int64)
+    result = np.zeros(n, dtype=np.int64)
     if _is_tzlocal(tz2):
```

Comment:
@kawochen : Hmmm...that would certainly explain why I was having a hard time reproducing it.  `np.empty` produces unpredictable values for `np.int64` and could easily produce the `NPY_NAT` value.  If all your tests pass, I'll close my PR.